### PR TITLE
Feature/cryptoservice transaction

### PR DESCRIPTION
**Pull request by mespsar**
Please review

### CODE REVIEW

**Review by bhautikgosai96 — DISMISSED** [review]
Hi Suryapal,

- Remove commented code if not require, system.out.println() and extra space in methods.
- Do proper indentation while importing packages and in if conditions.
- While inserting in database for example, save method handle exception as might get it  like (not null, unique constraint) and return error message for it.

File: src/main/java/com/pluang/OnboardingCryptoService/controllers/PortfolioController.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping(path="/portfolio")
+public class PortfolioController {
+
+    @Autowired
+    PortfolioService portfolioService;
+    //    get portfolio by id
+    @GetMapping(path="/portfolioget/{id}")
```
Comment (ankit2204) [inline src/main/java/com/pluang/OnboardingCryptoService/controllers/PortfolioController.java:14]: we already have portfolio in controller url mapping, we don't need portfolioget here.

File: src/main/java/com/pluang/OnboardingCryptoService/entities/Transaction.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    private Integer transactionid;
+
+    @Column(name="currency_id")
+    private Integer currencyid;
+
+    @Column(name="user_id")
+    private  Integer userid;
+
+    @Column(name="transaction_type")
+    private Boolean transactiontype;
```
Comment (ankit2204) [inline src/main/java/com/pluang/OnboardingCryptoService/entities/Transaction.java:30]: This should be enum or string, use of boolean for transaction_type is right

File: src/main/java/com/pluang/OnboardingCryptoService/services/AccountService.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    @Autowired
+    AccountRepository accountRepository;
+    //    get Account for balance
+    public Account getAccount(Integer id) {
+//        Optional<Account> account = accountRepository.findById(id);
+        Account account = accountRepository.getById(id);
+        return account;
+    }
+//    update balance
+    public Account putAccount(Account account) {
```
Comment (ankit2204) [inline src/main/java/com/pluang/OnboardingCryptoService/services/AccountService.java:20]: putAccount doesn't convey what this method is actually doing, use a better name like updateAccount

File: src/main/java/com/pluang/OnboardingCryptoService/services/PortfolioService.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+
+@Service
+public class PortfolioService {
+    @Autowired
+    PortfolioRepository portfolioRepository;
+    //update portfolio entry record
+    public Portfolio putPortfolio(Portfolio portfolio) {
```
Comment (ankit2204) [inline src/main/java/com/pluang/OnboardingCryptoService/services/PortfolioService.java:13]: use a better method name which convey what this method actually does

File: src/main/java/com/pluang/OnboardingCryptoService/services/PortfolioService.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        return portfolioRepository.save(portfolio);
+    }
+    public Portfolio getPortfolio(Integer id) {
+
+        return portfolioRepository.findById(id).orElse(null);
```
Comment (ankit2204) [inline src/main/java/com/pluang/OnboardingCryptoService/services/PortfolioService.java:18]: If we don't find the portfolio, we should throw a custom exception with 404 status code

**Review by ankit2204 — COMMENTED** [review]
Implementation is good, please follow Java naming conventions and try to improve code readability by indenting the code which is going too long in a single line.

**Pull request merged**